crates/viewer/re_data_ui/src/transform_frames.rs

```diff
@@ -0,0 +1,229 @@
+use re_chunk_store::UnitChunkShared;
+use re_log_types::EntityPath;
+use re_types::{
+    ComponentDescriptor, TransformFrameIdHash, archetypes,
+    components::{self, TransformFrameId},
+};
+use re_ui::{HasDesignTokens as _, UiExt as _, UiLayout, icons};
+use re_viewer_context::{
+    Item, SystemCommand, SystemCommandSender as _, TransformDatabaseStoreCache, ViewerContext,
+};
+
+/// The max amount of ancestors we show before putting a '…'.
+const MAX_SHOWN_ANCESTORS: usize = 100;
+
+/// The max amount of ancestors we show in a tooltip.
+///
+/// We want this to be less because we don't use a scroll area for
+/// tooltips.
+const MAX_SHOWN_ANCESTORS_TOOLTIP: usize = 3;
+
+struct TransformFrameInfo {
+    frame_id: TransformFrameId,
+    source_entity: Option<EntityPath>,
```

Comment:
should document that this is the entity that has connectects this frame_id with a parent frame. Got very confused on my reading and thought it was one of the entities having this frame as `CoordinateFrame`

also the `None` case is somewhat peculiar, as I understand so far that's really just an error case 🤔

Reply:
> also the None case is somewhat peculiar, as I understand so far that's really just an error case 🤔

In the `transform3d_hierarchy_frames` snippet, I get that `sun` is None.